hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/ozoneimpl/ContainerReader.java

```diff
@@ -148,7 +148,18 @@ public void readVolume(File hddsVolumeRootDir) {
       LOG.info("Start to verify containers on volume {}", hddsVolumeRootDir);
       File currentDir = new File(idDir, Storage.STORAGE_DIR_CURRENT);
       File[] containerTopDirs = currentDir.listFiles();
-      if (containerTopDirs != null) {
+      if (containerTopDirs != null && containerTopDirs.length > 0) {
+        try {
+          // idDir is working directory having data
+          // and volume is initialized with temp path
+          hddsVolume.createTmpDirs(idDir.getName());
```

Comment:
Thanks for the patch @sumitagrawl . Would it make sense to reuse StorageVolumeUtil.checkVolume here?

Reply:
ContainerReader thread is also run when ContainerCommands like `ozone debug container inspect|info|export|list` commands. Should we only check the volume and create tmp dirs during DN startup and not when container commands are run after the startup?
In that case, OzoneContainer.buildContainerSet() might be an ideal place to make this change.

Reply:
@SaketaChalamchala 
StorageVolumeUtil.checkVolume takes other action like, `upgradeVolume`, and `creates version files`, and create other directory. These action can not be taken at this point as` scm Id is not available` (only present after SCM registeration, and used to decide working directory or upgrade).

Here, it identifies working directory from directory path as existing code and using same for validation later. So trying initialize temp path in same working directory to have cleanup logic. Being temp directory, this will not have any impact.

`ozone debug container inspect|info|export|list` should not create temp dir, I think existing bug where it performs delete also. We should fix  using `shouldDeleteRecovering` for this flow. This needs fix.

Reply:
Updated avoiding deletion of container and its data, and also avoided creation of tmp folder.

Reply:
Makes sense @sumitagrawl. Just to clarify, with the `shouldDelete` fix temp dir is created and containers are removed only during startup and the other Container commands should not create/remove and dirs/containers.
The patch LGTM